inference-engine/thirdparty/clDNN/runtime/ocl/ocl_memory.cpp

```diff
@@ -306,11 +309,17 @@ gpu_usm::gpu_usm(ocl_engine* engine, const layout& layout, allocation_type type)
 }
 
 void* gpu_usm::lock(const stream& stream, mem_lock_type /*type*/) {
-    assert(get_allocation_type() != allocation_type::usm_device && "Can't lock usm device memory!");
     std::lock_guard<std::mutex> locker(_mutex);
     if (0 == _lock_count) {
-        stream.finish();  // Synchronization needed for OOOQ.
-        _mapped_ptr = _buffer.get();
+        auto& cl_stream = downcast<const ocl_stream>(stream);
+        cl_stream.finish();  // Synchronization needed for OOOQ.
+        if (get_allocation_type() == allocation_type::usm_device) {
+            _host_buffer.allocateHost(_bytes_count);
```

Comment:
As it is not expected scenario, I think we can inform about it in OV_GPU_Verbose mode

Reply:
Agree, Add copy to host buffer message in case OV_GPU_verbose=2.